python/ray/serve/tests/test_pipeline_driver.py

```diff
@@ -170,5 +171,34 @@ def test_driver_np_serializer(serve_instance):
     assert requests.get("http://127.0.0.1:8000/").json() == [42]
 
 
+def test_dag_driver_sync_warning(serve_instance, capsys):
+    with InputNode() as inp:
+        dag = echo.bind(inp)
+
+    handle = serve.run(DAGDriver.bind(dag))
+    assert ray.get(handle.predict.remote(42)) == 42
+
+    buffer = ""
+
+    def read_log():
+        nonlocal buffer
+
+        captured = capsys.readouterr()
+        buffer += captured.out
+        buffer += captured.err
+
+    def wait_for_request_success_log():
+        read_log()
+        lines = buffer.splitlines()
+        for line in lines:
+            if "DAGDriver" in line and "HANDLE predict OK" in line:
+                return True
+        return False
```

Comment:
you can simplify this a bit with `contextlib.redirect_stderr` -- see `test_logging.py` for usage